cards/portfolio/addon/components/isolated.js

```diff
@@ -1,10 +1,80 @@
 import LiveIsolatedCard from 'portfolio-common/components/live-isolated-card';
 import layout from '../templates/isolated';
+import { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+import { task, waitForProperty } from 'ember-concurrency';
 
 export default LiveIsolatedCard.extend({
   layout,
+  store: service(),
+  web3: service(),
+  erc20: service(),
   isDismissed: false,
   displayGroup: 'all',
+  metamaskWallet: null,
+
+  isWeb3Loaded: computed.not('web3.isLoading'),
+
+  async init() {
+    this._super(arguments);
+
+    if (this.web3.provider && this.web3.provider.isMetaMask) {
+      await this.getMetamaskWallet.perform();
+    }
+  },
+
+  getMetamaskWallet: task(function * () {
+    yield waitForProperty(this, 'isWeb3Loaded');
+
+    let address = this.web3.address;
+    let metamaskWallet = yield this.store.findRecord('wallet', 'metamask-wallet');
+
+    this.set('metamaskWallet', metamaskWallet);
+
+    yield this.getAssetsForWallet.perform(address);
+
+    this.content.get('wallets').pushObject(this.get('metamaskWallet'));
+
+    yield this.content.save();
+  }),
+
+  getAssetsForWallet: task(function * (address) {
+    yield this.addAssetToWallet.perform(address, 'ether');
+
+    for (let token of this.erc20.tokens()) {
+      let tokenName = token.symbol.toLowerCase();
+      let balanceOf;
+
+      try {
+        balanceOf = yield this.store.findRecord(`${tokenName}-token-balance-of`, address.toLowerCase());
+      } catch (e) {
+        return;
+      }
+
+      if (balanceOf) {
+        yield this.addAssetToWallet.perform(`${address}_${tokenName}-token`, tokenName);
+      }
+    }
+  }),
+
+  addAssetToWallet: task(function * (assetId, networkId) {
+    let asset;
+    let network = yield this.store.findRecord('network', networkId);
+
+    try {
+      asset = yield this.store.findRecord('asset', assetId);
+    } catch (e) {
+      asset = this.store.createRecord('asset', {
+        id: assetId,
+        network,
+        todaysRatesLookup: this.content.todaysRatesLookup
```

Comment:
i still dont understand the point of this field--its a server side computed, there is no need to set it.